File: 0526-beautiful-arrangement/0526-beautiful-arrangement.py
class Solution:
    def countArrangement(self, n: int) -> int:
        seen = [False for i in range (16)]
        # print(seen[10])
        res = 0
        def go(n,pos):
            nonlocal seen
            nonlocal res
            if(pos>n):
                res+=1
                print(res)
                return 
            for i in range(1,n+1):
                # print(seen[i])
                if(seen[i]==False and (i%pos==0 or pos%i==0)):
                    seen[i]=True
                    go(n,pos+1)
                    seen[i]=False
        if(n<4):
            return n
        go(n,1)
        return res

File: 0526-beautiful-arrangement/0526-beautiful-arrangement.cpp
class Solution {
public:
    bool seen[16] = {};
    int res =0;
    void go(int n,int pos){
        if(pos>n){
            res+=1;
        }
        for(int i=1;i<=n;++i){
            if(seen[i]==false and (i%pos==0 or pos%i==0)){
                seen[i]=1;
                go(n,pos+1);
                seen[i]=0;
            }
        }
    }
    int countArrangement(int n) {
        if(n<4) return n;
        go(n,1);
        return res;
    }
};